downloads/swan-lake-release-notes/swan-lake-alpha/RELEASE_NOTE.md

```diff
@@ -0,0 +1,1432 @@
+
+###### Client Changes
+
+- A new `sendEmail` method is added to the `email:SmtpClient` API to send emails directly without creating the `email:Email` record supporting extra fields as named optional parameters. An example of sending an email with the new API is given below.
+
+    ```ballerina
+    Error? response = smtpClient->sendEmail(toAddresses, subject, fromAddress, body, cc = ccAddresses, bcc = bccAddresses, 
+    htmlBody = htmlBody, contentType = contentType, headers = {header1_name: "header1_value"}, sender = sender, replyTo = 
+    replyToAddresses, attachments = bodyParts);
+    ```
+
+- The `read` method of `email:ImapClient`, `email:PopClient`, and `email:Listener` (i.e., `new email:PopListener` and `email:ImapListener`) are changed to `receiveEmailMessage`.
+
+- All client initializations return an `email:Error` when an error occurs during the initialization.
+
+###### Service and Listener Related Changes
+
+- All listener initializations return an `email:Error` when an error occurs during the initialization.
+
+- The `email:Listener` is split into the `email:PopListener` and `email:ImapListener`. Therefore, the `protocol` field is removed from the new protocol-specific listeners. The `email:PopConfig` or `email:ImapConfig` that were used as fields for the `email:Listener` are not required for the new API implementation. Protocol configuration related fields are made parts of the new listeners.
+
+- The `resource` functions are changed to `remote` functions in the new listener APIs.
+
+- The `service` name is given as a string with the new Ballerina language changes.
+
+- The `onMessage` method of the `email:Listener` (i.e., `new email:PopListener` and `email:ImapListener`) are changed to `onEmailMessage`.
+
+- The `pollingInterval` field of the `email:Listener` is changed to `pollingIntervalInMillis` in the new listener APIs. That makes it consistent across other Ballerina modules, which are time durations configured in milliseconds.
+
+- A sample POP3 listener is given below.
+
+    **New Syntax:**
+
+    ```ballerina
+    listener email:PopListener emailListener = new ({
+        host: "pop.email.com",
+        username: "reader@email.com",
+        password: "pass456",
+        pollingIntervalInMillis: 2000,
+        port: 995
+    });
+
+    service "emailObserver" on emailListener {
+
+    remote function onEmailMessage(email:Message emailMessage) {
+
+        }
+
+    remote function onError(email:Error emailError) {
+
+        }
+    }
+    ```
+
+##### Introduced Modules
+
+###### Random
+
+This module provides functions related to random number generation.
+
```

Comment:
Regex module has not been added under the list of introduced modules.

Reply:
Added.